pyvista/utilities/reader.py

```diff
@@ -0,0 +1,639 @@
+        """Return the status of all point arrays.
+        
+        Returns
+        -------
+        dict[str, bool]
+
+        """
+        return {name: self.point_array_status(name) for name in self.point_array_names}
+
+    @property
+    def number_cell_arrays(self):
+        """Return the number of cell arrays.
+        
+        Returns
+        -------
+        int
+
+        """
+        return self.reader.GetNumberOfCellArrays()
+
+    @property
+    def cell_array_names(self):
+        """Return the list of all cell array names.
+        
+        Returns
+        -------
+        list[str]
+
+        """
+        return [self.reader.GetCellArrayName(i) for i in range(self.number_cell_arrays)]
+
+    def enable_cell_array(self, name):
+        """Enable cell array with name.
+
+        Parameters
+        ----------
+        name: str
+            Cell array name
+
+        """
+        self.reader.SetCellArrayStatus(name, 1)
+
+    def disable_cell_array(self, name):
+        """Disable cell array with name.
+
+        Parameters
+        ----------
+        name: str
+            Cell array name
+
+        """
+        self.reader.SetCellArrayStatus(name, 0)
+
+    def cell_array_status(self, name):
+        """Get status of cell array with name.
+
+        Parameters
+        ----------
+        name: str
+
+        Returns
+        -------
+        bool
+
+        """
+        if self.reader.GetCellArrayStatus(name):
+            return True
+        return False
+
+    def enable_all_cell_arrays(self):
+        """Enable all cell arrays."""
+        for name in self.cell_array_names:
+            self.enable_cell_array(name)
+
+    def disable_all_cell_arrays(self):
+        """Disable all cell arrays."""
+        for name in self.cell_array_names:
+            self.disable_cell_array(name)
+
+    @property
+    def all_cell_arrays_status(self):
+        """Return the status of all cell arrays.
+        
+        Returns
+        -------
+        dict[str, bool]
+        """
+        return {name: self.cell_array_status(name) for name in self.cell_array_names}
+
+
+class XMLImageDataReader(BaseReader, PointCellDataSelection):
+    """XML Image Data Reader."""
+
+    _class_reader =_vtk.vtkXMLImageDataReader
+
+
+class XMLPImageDataReader(BaseReader, PointCellDataSelection):
+    """XML P Image Data Reader."""
+
+    _class_reader = _vtk.vtkXMLPImageDataReader
+
+    
+class XMLRectilinearGridReader(BaseReader, PointCellDataSelection):
+    """XML RectilinearGrid Reader.
+    
+    Examples
+    --------
+    >>> import pyvista
+    >>> from pyvista import examples
+    >>> filename = examples.download_rectilinear_grid(load=False)
+    >>> print(filename.split("/")[-1])  # omit the path
+    RectilinearGrid.vtr
+    >>> reader = pyvista.get_reader(filename)
+    >>> mesh = reader.read()
+    >>> sliced_mesh = mesh.slice('y')
+    >>> sliced_mesh.plot(scalars="Void Volume Fraction", cpos='xz', show_scalar_bar=False)
```

Comment:
I originally had this as 

```py
>>> sliced_mesh.plot(scalars="Void Volume Fraction",  # Doc building complained of EOL here
...                  cpos='xz', show_scalar_bar=False)

```

This passed doctest locally, but failed in building the documentation.  It looks a little long to me on one line, but I couldn't figure out how to break itup.

Reply:
This sounds really weird. We have a bunch of these elsewhere in the docs. E.g. https://github.com/pyvista/pyvista/blob/fab7f777aa4af5d9007cef12c77738236e66e77e/pyvista/core/filters/data_set.py#L2842-L2844 rendered at https://dev.pyvista.org/api/core/_autosummary/_autosummary/pyvista.DataSetFilters.streamlines_evenly_spaced_2D.html#pyvista.DataSetFilters.streamlines_evenly_spaced_2D.

Are you sure there wasn't some other typo there? Missing comma or something like that?

Reply:
:shrug:  I pushed the version that fails in the documentation but passes in the doctest as it will probably be more helpful for debugging.

Maybe it is the way I run doctest???  I am using `pytest --doctest-modules pyvista/utilities/reader.py`

Reply:
It passed here, but fails on my local doc build.  Curious.  Maybe it is a dependency version thing?